nomad/structs/config/tls.go

```diff
@@ -1,7 +1,14 @@
 package config
 
+import (
+	"crypto/tls"
+	"fmt"
+	"sync"
+)
+
 // TLSConfig provides TLS related configuration
 type TLSConfig struct {
+	configLock sync.Mutex
```

Comment:
It looks like this only guards KeyLoader, so perhaps rename it `keyLoaderLock` and put the field below keyloader (we try to put lock field declarations near what they lock).